showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,517 @@
+
+public class ITOtelMetrics {
+
+  @AfterClass
+  public static void cleanup_otelcol() throws Exception {
+    Process process = Runtime.getRuntime().exec("../scripts/cleanup_otelcol.sh");
+    process.waitFor();
+  }
+
+  // This test is currently giving an error about requestId. will ask Alice and resolve it.
+  // @Test
+  // public void testHttpJson_OperationSucceded_recordsMetrics() throws Exception {
+  //   generate_otelcol_config("4317",
+  // "../test_data/testHttpJson_OperationSucceeded_metrics.txt","../test_data/testHttpJson_OperationSucceeded.yaml");
+  //   // initialize the otel-collector
+  //   setupOtelCollector("../test_data/testHttpJson_OperationSucceeded.yaml");
+  //
+  //   EchoClient client =
+  //       TestClientInitializer.createHttpJsonEchoClientOpentelemetry(
+  //           createOpenTelemetryTracerFactory("4317"));
+  //
+  //   EchoRequest requestWithNoError =
+  //       EchoRequest.newBuilder()
+  //           .setContent("successful httpJson request")
+  //           .build();
+  //
+  //   client.echo(requestWithNoError);
+  //
+  //   // wait for the metrics to get uploaded
+  //   Thread.sleep(5000);
+  //
+  //   String filePath = "../test_data/testHttpJson_OperationSucceeded_metrics.txt";
+  //   String attribute1 =
+  //
+  // "\"attributes\":[{\"key\":\"language\",\"value\":{\"stringValue\":\"Java\"}},{\"key\":\"method_name\",\"value\":{\"stringValue\":\"google.showcase.v1beta1.Echo/Echo\"}},{\"key\":\"status\",\"value\":{\"stringValue\":\"OK\"}}]";
+  //
+  //   String[] params = {filePath, attribute1};
+  //   int result = verify_metrics(params);
+  //   Truth.assertThat(result).isEqualTo(0);
+  //
+  //   client.close();
+  //   client.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  // }
+
+  @Test
+  public void testGrpc_OperationSucceded_recordsMetrics() throws Exception {
+
+    generate_otelcol_config(
+        "4318",
+        "../test_data/testGrpc_OperationSucceeded_metrics.txt",
+        "../test_data/testGrpc_OperationSucceeded.yaml");
+    setupOtelCollector("../test_data/testGrpc_OperationSucceeded.yaml");
+
+    EchoClient client =
+        TestClientInitializer.createGrpcEchoClientOpentelemetry(
+            createOpenTelemetryTracerFactory("4318"));
+
+    EchoRequest requestWithNoError =
+        EchoRequest.newBuilder().setContent("test_grpc_operation_succeeded").build();
+
+    client.echo(requestWithNoError);
+
+    // wait for the metrics to get uploaded
+    Thread.sleep(5000);
+
+    String filePath = "../test_data/testGrpc_OperationSucceeded_metrics.txt";
+    String attribute1 =
+        "\"attributes\":[{\"key\":\"language\",\"value\":{\"stringValue\":\"Java\"}},{\"key\":\"method_name\",\"value\":{\"stringValue\":\"Echo.Echo\"}},{\"key\":\"status\",\"value\":{\"stringValue\":\"OK\"}}]";
+
+    String[] params = {filePath, attribute1};
+    int result = verify_metrics(params);
+    Truth.assertThat(result).isEqualTo(0);
```

Comment:
I'm not sure that we need a separate shell script to do the verification. This looks JSON to me and I think you can just read the file and parse it with a JSON library (Gson for example). Validation can be used with Json.

Was there an issue you were facing regarding using Gson?